### ISSUE TITLE
More customization to the menu

### ISSUE THREAD

**Issue by lojack** [wagtail/wagtail#359]
I'd like the ability to have more control over the admin menu's.

A Specific use case would be to only show a subpage (Blog index), Images, and Documents in the navigation for Blog Editors. A hacky way I can accomplish this right now would be to clear the menu then manually add what I want, and there's obvious reasons why this is a bad idea...

``` python
def construct_main_menu(request, menu_items):
    page = Page.objects.filter(slug="blog").first()
    if page:
        menu_items[:] = []
        menu_items.append(
            MenuItem(_('Blog'), urlresolvers.reverse('wagtailadmin_explore', args=(page.id,)), classnames='icon icon-doc-full', order=101)
        )

hooks.register('construct_main_menu', construct_main_menu)
```

There's two potential changes that could accomplish what I need. 

The first would be to make Explorer and Search [not be hard coded](https://github.com/torchbox/wagtail/blob/58ed9181290b1aba1da652e3d0dbbb12eef444d6/wagtail/wagtailadmin/templatetags/wagtailadmin_tags.py#L31) into `main_nav`, and have some sort of permission to control how it shows in the menu -- [like in the Users hook](https://github.com/torchbox/wagtail/blob/79334134b79adc1f79acdede3da3a17555a170db/wagtail/wagtailusers/wagtail_hooks.py#L19).

Alternatively, the ability to set an `admin_root` on either users or groups, and setting the explorer root to this Page along with filtering search results to only include these pages.

**Comment by gasman** [comment]
A status update on this... Point 2 (restricting the navigation to the smallest subtree that covers all the user's page permissions) was implemented in #2869, and point 1 will be addressed by #3068 (currently awaiting review).

**Comment by gasman** [comment]
Completed in #3068
